python/python/lance/file.py

```diff
@@ -0,0 +1,188 @@
+#  Copyright (c) 2023. Lance Developers
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from typing import Union
+
+import pyarrow as pa
+
+from .lance import (
+    LanceBufferDescriptor,
+    LanceColumnMetadata,
+    LanceFileMetadata,
+    LancePageMetadata,
+)
+from .lance import (
+    LanceFileReader as _LanceFileReader,
+)
+from .lance import (
+    LanceFileWriter as _LanceFileWriter,
+)
+
+
+class ReaderResults:
+    """
+    Utility class for converting results from Lance's internal
+    format (RecordBatchReader) to a desired format such
+    as a pyarrow Table, etc.
+    """
+
+    def __init__(self, reader: pa.RecordBatchReader):
+        """
+        Creates a new instance, not meant for external use
+        """
+        self.reader = reader
+
+    def to_batches(self) -> pa.RecordBatchReader:
+        """
+        Return the results as a pyarrow RecordBatchReader
+        """
+        return self.reader
+
+    def to_table(self) -> pa.Table:
+        """
+        Return the results as a pyarrow Table
+        """
+        return self.reader.read_all()
+
+
+class LanceFileReader:
+    """
+    A file reader for reading Lance files
```

Comment:
Is this interface mostly internal utility? The public main interface is still `pyarrow.Dataset`?

Reply:
Yes.  This is mainly for benchmarking / advanced use cases.